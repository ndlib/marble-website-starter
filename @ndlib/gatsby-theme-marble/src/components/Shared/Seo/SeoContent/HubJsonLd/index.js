import React from 'react'
import PropTypes from 'prop-types'
import { JsonLd } from '../JsonLd'

export const HubJsonLd = (props) => {
  const {
    thumbnail,
    pathname,
    title,
    description,
    date,
    url,
    image,
    language,
  } = props
  if (pathname.includes('/featured/')) {
    return (
      <JsonLd>
        {{
          '@context': 'https://schema.org',
          '@type': 'CollectionPage',
          mentions: {
            '@type': 'SoftwareApplication',
            operatingSystem: 'All',
            applicationCategory: 'BusinessApplication',
          },
          about: description,
          name: title,
          url: url,
          dateCreated: date,
          inLanguage: language,
          image: image,
          thumbnailUrl: thumbnail,
        }}
      </JsonLd>
    )
  }
  return null
}

HubJsonLd.propTypes = {
  pathname: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  date: PropTypes.string,
  image: PropTypes.string.isRequired,
  thumbnail: PropTypes.string.isRequired,
  language: PropTypes.string,
}

export default HubJsonLd
