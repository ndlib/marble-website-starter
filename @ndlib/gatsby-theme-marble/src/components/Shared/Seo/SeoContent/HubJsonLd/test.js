import React from 'react'
import renderer from 'react-test-renderer'
import HubJsonLd from './'

describe('HubJsonLd component', () => {
  it('renders props into values', () => {
    const tree = renderer
      .create(<HubJsonLd
        pathname={'/featured/posters'}
        title={'1984'}
        image={'http://fake.io/image'}
        thumbnail={'http://fake.io/thumbnail'}
        url={'http://object.io/page'}
        date={'Present'}
        description={'Dystopian nightmare'}
        language={'Newspeak'}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
