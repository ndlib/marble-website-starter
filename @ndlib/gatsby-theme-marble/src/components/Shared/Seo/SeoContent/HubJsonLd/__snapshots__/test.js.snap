// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubJsonLd component renders props into values 1`] = `null`;
